### PR TITLE
GPII-3853: Analysis related metrics improvements (new)

### PR DESCRIPTION
**Pull request by stegru**
A remake of https://github.com/GPII/gpii-app/pull/139, but without any extra crap.

More info: https://github.com/GPII/universal/pull/790

### CODE REVIEW

**Comment by stegru** [comment]
new one:

> Not enough memory resources are available to complete this operation.

```
    default: V:\node_modules\@pokusew\pcsclite>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild ) 
    default: Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
    default:   addon.cpp
    default:   pcsclite.cpp
An error occurred executing a remote WinRM command.

Shell: Elevated
Command: powershell -ExecutionPolicy Bypass -OutputFormat Text -file "c:\tmp\vagrant-shell.ps1" -originalBuildScriptPath "C:\vagrant\provisioning\"
Message: [WSMAN ERROR CODE: 14]: <f:WSManFault Code='14' Machine='127.0.0.1' xmlns:f='http://schemas.microsoft.com/wbem/wsman/1/wsmanfault'><f:Message>Not enough memory resources are available to complete this operation. </f:Message></f:WSManFault>
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```

**Comment by stegru** [comment]
ok to test

File: provisioning/service.json5 [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            // Get the URL from the secrets file
+            url: "${siteConfigUrl}"
+        }],
+    },
+    "siteConfigFile": [
+        "%ProgramData%\\Morphic\\siteConfig.json5",
+        "%ProgramFiles(x86)%\\Morphic\\windows\\resources\\app\\siteConfig.json5",
+        "%ProgramFiles%\\Morphic\\windows\\resources\\app\\siteConfig.json5"
+    ],
+    "gpii-config": {
```
Comment (javihernandez) [inline provisioning/service.json5]: better to use gpiiConfig

Comment (stegru) [reply provisioning/service.json5]: Well spotted.

I did, but I wrote this file before the implementation ;)

**Comment by stegru** [comment]
ok to test

**Comment by javihernandez** [comment]
@Karadaliev after this PR is updated with current master, would you mind giving this a review?

**Comment by Karadaliev** [comment]
> @Karadaliev after this PR is updated with current master, would you mind giving this a review?

Sure, I will look for changes and when @stegru merge everything will go through his code and look for stuff I understand 🛩️ .

**Comment by javihernandez** [comment]
ok to test

**Comment by stegru** [comment]
ok to test

**Comment by amb26** [comment]
Test case hang in Widget integration tests
````
04:55:48.039:  Test case listener has not responded after 5000ms - at sequence pos 230 of 257 sequence element {
    "event": "{that}.app.qssWrapper.qssWidget.events.onQssWidgetCreated",
    "listener": "fluid.identity"
} of fixture QSS Widget integration tests
````
ok to test

**Comment by javihernandez** [comment]
ok to test

**Comment by amb26** [comment]
Standard lost widget integration test failure - ok to test

**Comment by amb26** [comment]
Same again:
````
10:17:24.439:  RendererCoverage: Still awating: [
    "gpii.app.qssWidget"
]
````
ok to test

**Comment by amb26** [comment]
First kind of failure again:
````
06:58:02.306:  Test case listener has not responded after 5000ms - at sequence pos 230 of 257 sequence element {
    "event": "{that}.app.qssWrapper.qssWidget.events.onQssWidgetCreated",
    "listener": "fluid.identity"
} of fixture QSS Widget integration tests
````
ok to test

**Comment by amb26** [comment]
Got that locally too

File: src/renderer/common/js/metrics.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        },
+        invokers: {
+            getMetricsID: {
+                funcName: "gpii.psp.metrics.getWidgetMetricsID",
+                args: ["{that}"]
+            }
+        },
+        listeners: {
+            "onQssWidgetCreated.metric": {
+                funcName: "gpii.psp.metrics.addTipLinkHandlers",
```
Comment (amb26) [inline src/renderer/common/js/metrics.js:119]: Is it possible this listener is throwing, preventing the tests passing?

**Comment by amb26** [comment]
Test failure is probably 'Click on the "Adjust Mouse" button in qssWidget-mouseTests.js

File: src/renderer/common/js/metrics.js (hunk 2 above)
Comment (amb26) [reply src/renderer/common/js/metrics.js:119]: Tried commenting this out but tests still fail

**Comment by amb26** [comment]
Same failure as before:
````
08:30:42.140:  FactsManager: Changed interactionsCount: 23
08:30:47.152:  Test case listener has not responded after 5000ms - at sequence pos 230 of 257 sequence element {
    "event": "{that}.app.qssWrapper.qssWidget.events.onQssWidgetCreated",
    "listener": "fluid.identity"
} of fixture QSS Widget integration tests
````

**Comment by kaspermarkus** [comment]
YAY!

**Pull request merged**